src/Command/DetermineConfigurationFromFilesystem.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Roave\BackwardCompatibility\Command;
+
+use Psl\Str;
+use Psl\Type;
+use Roave\BackwardCompatibility\Configuration\Configuration;
+use Roave\BackwardCompatibility\Configuration\ParseConfigurationFile;
+use Roave\BackwardCompatibility\Configuration\ParseXmlConfigurationFile;
+use Symfony\Component\Console\Output\OutputInterface;
+
+final class DetermineConfigurationFromFilesystem
+{
+    public function __construct(
+        private readonly ParseConfigurationFile $parser = new ParseXmlConfigurationFile(),
```

Comment:
Overall OK to do this, but it's gonna be fun to see all tooling break because of our first time using objects as default values 😁

Reply:
Why would it break? Unsupported?

Reply:
Possibly in tooling, even in here. @DanielBadura has sent patches here to fix compatibility around that before, but it's really recent stuff :)